ui/app/templates/components/job-editor.hbs

```diff
@@ -1,156 +1,147 @@
-          mode="javascript"
-        }}
-      />
+<div>
+  {{#if this.error}}
+    <div data-test-error={{this.error.type}} class="notification is-danger">
+      <h3 class="title is-4" data-test-error-title>{{conditionally-capitalize this.error.type true}} Error</h3>
+      <p data-test-error-message>{{this.error.message}}</p>
     </div>
-  </div>
+  {{/if}}
+  {{#if (eq this.stage "editor")}}
 
-  <div class="is-associative buttonset">
-    <Hds::Button
-      {{on "click" (perform this.plan)}}
-      disabled={{or this.plan.isRunning (not this.job._newDefinition)}}
-      data-test-plan
-      @text="Plan"
-    />
-    <Hds::ButtonSet>
-      <label class="job-spec-upload hds-button hds-button--color-secondary hds-button--size-medium">
-        <div class="hds-button__text">Upload file</div>
-        <input type="file" onchange={{action this.uploadJobSpec}} accept=".hcl,.json,.nomad" />
-      </label>
-      {{#if (can "write variable" path="*" namespace="*")}}
-        <Hds::Button @icon="file-plus" @text="Save as template" @color="tertiary" @route="jobs.run.templates.new"  {{on "click" this.handleSaveAsTemplate}} data-test-save-as-template />
-        <Hds::Button @icon="file-text" @text="Choose from a template" @color="tertiary" @route="jobs.run.templates" data-test-choose-template />
-      {{/if}}
-    </Hds::ButtonSet>
-  </div>
-{{/if}}
+    <header class="run-job-header">
+      <h1 class="title is-3">Run a job</h1>
+      <p>
+        Paste or author HCL or JSON to submit to your cluster, or select from a list of templates. A plan will be requested before the job is submitted. You can also attach a job spec by uploading a job file or dragging &amp; dropping a file to the editor.
+      </p>
+    </header>
 
-{{#if (eq this.stage "plan")}}
-  {{#if this.showPlanMessage}}
-    <div class="notification is-info">
-      <div class="columns">
-        <div class="column">
-          <h3 class="title is-4" data-test-plan-help-title>Job Plan</h3>
-          <p data-test-plan-help-message>This is the impact running this job
-            will have on your cluster.</p>
-        </div>
-        <div class="column is-centered is-minimum">
-          <Hds::Button @text="Okay" {{on "click" (toggle-action "showPlanMessage" this)}} data-test-plan-help-dismiss />
-        </div>
+    <div class="boxed-section">
+      <div class="boxed-section-head">
+        Job Definition
+        {{#if @cancelable}}
+          <button
+            class="button is-light is-compact pull-right"
+            onclick={{action @onCancel}}
+            data-test-cancel-editing
+            type="button"
+          >Cancel</button>
+        {{/if}}
+      </div>
+      <div class="boxed-section-body is-full-bleed">
+        <div
+          data-test-editor
+          {{code-mirror
+            screenReaderLabel="Job definition"
+            content=@job._newDefinition
+            theme="hashi"
+            onUpdate=this.updateCode
+            mode="javascript"
```

Comment:
Not for this PR but elsewhere: this should be conditional. Even today we would benefit from it being `javascript` on the definition page and `ruby` on the run index page.